backend/tests/routes/access/access.test.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * @jest-environment node
+ */
+import router from '../../../src/routes/access/access';
+
+describe('access router', () => {
+  const routes = [
+    { path: '/journal/:journalId', method: 'put' },
+    { path: '/login', method: 'post' },
+    { path: '/token-login', method: 'post' },
+    { path: '/forgot-password', method: 'post' },
+    { path: '/reset-password', method: 'post' },
+    { path: '/logout', method: 'get' },
+    { path: '/register', method: 'post' },
+    { path: '/verify-email', method: 'post' },
+    { path: '/beta-approval', method: 'get' },
+    { path: '/beta-denial', method: 'get' },
+  ];
+
+  it.each(routes)('should have all routes defined in access.ts', (route) => {
```

Comment:
I tried using ts-jest for type checking and it broke this test. The interface of the routers.stack[i].route object doesn't have a .methods property, but it has it at runtime for some reason. I think it only passes because babel can convert typescript, but doesn't do type checking.

For something quick and simple, I think these tests could be rewritten to use supertest to send a request to all of the endpoints and check that it doesn't get a 404 error back.

Also, using ts-jest didn't fix the issues with importing modules from the index.js barrel file. For now I think we can keep going

Reply:
This [PR](https://github.com/hiyaryan/the-cdj/issues/156) should hopefully resolve this issue. It doesn't use `ts-jest` and it also resolves some Jest setup and teardown issues that conflict with `mongodb-memory-server`.